src/components/Filters/hooks/useStateButton.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+import { StateButtonIconHook } from './useStateButton.types';
+import { SSCIconNames } from '../../Icon/Icon.enums';
+import { Color } from '../../../theme/colors.types';
+import { ColorTypes } from '../../../theme/colors.enums';
+
+const timesIconColor = ColorTypes.graphite2B;
+const checkIconColor = ColorTypes.graphiteB;
+const hoverIconColor = ColorTypes.graphite5H;
+
+export const useStateButtonIcon = (
+  isFilterApplied: boolean,
+): StateButtonIconHook => {
+  const [iconName, setIconName] = useState<string>(SSCIconNames.times);
+  const [iconColor, setIconColor] = useState<Color>(timesIconColor);
+
+  useEffect(() => {
+    if (isFilterApplied) {
+      setIconName(SSCIconNames.check);
+      setIconColor(checkIconColor);
+    }
+  }, [isFilterApplied]);
+
+  const handleMouseOut = () => {
```

Comment:
These `handleMouseOut` and `handleMouseOver` methods probably should be wrapped in `useCallback` since you are passing them into `RemoveButton` component. @abykovssc could you please confirm I'm still not 100% sure how this should be used :D

Reply:
No need to wrap it imo, since this hook is used in one place only (as i understand it isn't some kind of public api), and RemoveButton isn't memoized, so wrapping this handlers into `useCallback` isn't going to achieve anything